src/assisted_test_infra/test_infra/controllers/minikube.py

```diff
@@ -0,0 +1,242 @@
+        :param replicas_count: total requested replicas to verify in the spec
+        :param timeout: 10 second default
+        :param sleep: 2 second interval
+        :return:
+        """
+        waiting.wait(
+            lambda: self._get_obj_resources_by_name(resource_kind_name, name)[0].spec.replicas == replicas_count,
+            timeout_seconds=timeout,
+            sleep_seconds=sleep,
+            waiting_for=f"replicas to change successfully to {replicas_count}",
+        )
+
+    def wait_for_pod_change(self, name, replicas_count, timeout=180, sleep=10, recover_time_pods=20):
+        """Verify pod created / deleted based on replicas numbers
+        In case we create pods (replicas) need to wait for a running state
+        :param name: lookup name begins with assisted-service
+        :param replicas_count: total number of replicas
+        :param timeout: totl wait time till pods deleted or created ready
+        :param sleep: sleep between interval
+        :param recover_time_pods: after stop/start pod may take more time to load
+        :return:
+        """
+        waiting.wait(
+            lambda: len(self._get_obj_resources_by_name(self.POD, name)) == replicas_count,
+            timeout_seconds=timeout,
+            sleep_seconds=sleep,
+            waiting_for=f"replicas to change successfully to {replicas_count} for {name}",
+        )
+        # In case the scale increased check if all pods are ready and active
+        if replicas_count:
+
+            def pods_are_running():
+                # Waiter lambda verify all pods are running begins with the same name
+                pods = self._get_obj_resources_by_name(self.POD, name)
+                running = list(map(lambda pod: hasattr(pod.status, "phase") and pod.status.phase == "Running", pods))
+                return all(running)
+
+            waiting.wait(
+                lambda: pods_are_running() is True,
+                timeout_seconds=timeout,
+                sleep_seconds=sleep,
+                waiting_for="replicas to change successfully",
+            )
+        time.sleep(recover_time_pods)
+
+    def change_deployment_scale(self, deployment_name, replicas_count):
+        """Common code for scale up and down deployment
+        We assume that assisted-service name is unique in deployment. other kinds
+        name starts with same name with uuids like assisted-service-9fdd7f47-xd8vz
+        :param deployment_name: assisted-service
+        :param replicas_count: total number , 0 for stop 1 and more for start
+        :return:
+        """
+        deployment = self._get_obj_resources_by_name(self.DEPLOYMENT, deployment_name)
+        # verify single name for deployment replicaset exists
+        assert len(deployment) == 1, "deployment kind name should be unique"
+        replicaset = self._get_obj_resources_by_name(self.REPLICASET, deployment_name)
+        assert len(replicaset) == 1, "replicaset kind name should be unique"
```

Comment:
Maybe create a function for getting deployment/replicaset/configmap..?
it will be clearer if we will not see the assert in the middle of the flow, e.g.
```python
deployment = self._get_resource(self.DEPLOYMENT, deployment_name)
replicaset = self._get_resource(self.REPLICASET, deployment_name)
```

While the method will look something like that:
```python
def _get_resource(resource_type: Resource, resource_name: str):
    resource = self._get_obj_resources_by_name(resource_type.value, resource_name)
    assert len(replicaset) == 1, f"{resource_type.value} kind name should be unique, name={resource_name}"
    return resource
```

Reply:
ack , done